lib/cocoapods/target/pod_target.rb

```diff
@@ -216,70 +216,52 @@ def supported_test_types
+            framework = { :name => framework_path.basename.to_s,
+                          :input_path => "${PODS_ROOT}/#{relative_path_to_sandbox}",
+                          :output_path => "${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/#{framework_path.basename}" }
+            # Until this can be configured, assume the dSYM file uses the file name as the framework.
+            # See https://github.com/CocoaPods/CocoaPods/issues/1698
+            dsym_name = "#{framework_path.basename}.dSYM"
+            dsym_path = Pathname.new("#{framework_path.dirname}/#{dsym_name}")
+            if dsym_path.exist?
+              framework[:dsym_name] = dsym_name
+              framework[:dsym_input_path] = "${PODS_ROOT}/#{relative_path_to_sandbox}.dSYM"
+              framework[:dsym_output_path] = "${DWARF_DSYM_FOLDER_PATH}/#{dsym_name}"
+            end
+            frameworks << framework
           end
-          frameworks << framework
-        end
-        if should_build? && requires_frameworks? && !static_framework?
-          frameworks << { :name => product_name,
-                          :input_path => build_product_path('${BUILT_PRODUCTS_DIR}'),
-                          :output_path => "${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/#{product_name}" }
+          if !file_accessor.spec.test_specification? && should_build? && requires_frameworks? && !static_framework?
+            frameworks << { :name => product_name,
+                            :input_path => build_product_path('${BUILT_PRODUCTS_DIR}'),
+                            :output_path => "${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/#{product_name}" }
+          end
+          hash[file_accessor.spec.name] = frameworks
         end
-        frameworks
       end
     end
 
-    # Returns the resource paths associated with this target. By default all paths include the resource paths
-    # that are part of test specifications.
+    # @return [Hash{String=>Array<String>}] The resource and resource bundle paths this target depends upon keyed by
+    #         spec name.
     #
-    # @param  [Boolean] include_test_spec_paths
-    #         Whether to include resource paths from test specifications or not.
-    #
-    # @return [Array<String>] The resource and resource bundle paths this target depends upon.
-    #
-    def resource_paths(include_test_spec_paths = true)
-      @resource_paths ||= {}
-      return @resource_paths[include_test_spec_paths] if @resource_paths.key?(include_test_spec_paths)
-      @resource_paths[include_test_spec_paths] = begin
-        accessors = file_accessors
-        accessors = accessors.reject { |a| a.spec.test_specification? } unless include_test_spec_paths
-        resource_paths = accessors.flat_map do |accessor|
-          accessor.resources.flat_map { |res| "${PODS_ROOT}/#{res.relative_path_from(sandbox.project.path.dirname)}" }
-        end
-        resource_bundles = accessors.flat_map do |accessor|
```

Comment:
this was stupid as it was looping twice over the `file_accessors` . We only need to loop once to get both `resources` and `resource_bundles` paths